libs/filer/fake_filer.go

```diff
@@ -0,0 +1,134 @@
+package filer
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"path"
+	"sort"
+	"strings"
+	"time"
+)
+
+type FakeDirEntry struct {
+	FakeFileInfo
+}
+
+func (entry FakeDirEntry) Type() fs.FileMode {
+	typ := fs.ModePerm
+	if entry.FakeDir {
+		typ |= fs.ModeDir
+	}
+	return typ
+}
+
+func (entry FakeDirEntry) Info() (fs.FileInfo, error) {
+	return entry.FakeFileInfo, nil
+}
+
+type FakeFileInfo struct {
+	FakeName string
+	FakeSize int64
+	FakeDir  bool
+	FakeMode fs.FileMode
```

Comment:
Given that the filer is called fake already the prefix can be omitted from the field names for brevity, IMO.